### PR TITLE
feat(notification): support role prop

### LINKED ISSUE #39568: Notification component's role="alert" creates accessibility issue for screen reader users
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/e9evzr)

### Steps to reproduce

1. Open example.
2. Click button to open notification box.

### What is actually happening?

Since `role="alert"` is used, the notification interrupt whatever the user is doing. This also stops a screen reader from finishing whatever it was saying before the alert interrupted it (and will not pick up where it left off) which can make a user feel stranded. Toast-style notifications should be used for non-critical, non-interactive content so it's inappropriate to use `role="alert"` which implicitly has an `aria-live` value of `assertive`.

### What is expected?

Instead, `role="status"` is more appropriate as it allows for current screen reader announcements to be read before announcing the non-critical notification content. This requires the notification elements to be with an element with `role="region"` and `aria-live="polite"`.

| Environment | Info |
| --- | --- |
| antd | 4.24.5 |
| React | React |
| System | Windows 10 |
| Browser | All |

---
More information:
- https://www.scottohara.me/blog/2019/07/08/a-toast-to-a11y-toasts.html#inclusive-ux-of-a-toast
- https://adrianroselli.com/2020/01/defining-toast-messages.html#NeededFeatures

### PR DESCRIPTION
**Pull request by guan404ming**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

1. close #39568 

### 💡 Background and solution

- ref: #39611

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Adds role props to Notification    |
| 🇨🇳 Chinese |     添加 role props 到 Notification     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f7986c4</samp>

Add `role` prop to notification component to improve accessibility. The prop allows users to specify the semantics of the notification content for screen readers, and is supported by the `useNotification` hook, the `PurePanel` component, and the documentation files.

### <samp>🤖 Generated by Copilot at f7986c4</samp>

*  Add `role` prop to notification component to support screen reader semantics ([link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-c80bd08ac7a1f6a8ae2046fe84cc562807b2d11b1a63316286546b888a86ed08R32), [link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR40), [link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfL108-R108), [link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-fa8b6f08e906ed18759c056ccb358fb0b780b4ffbb0f00ed368b45c3b8efe5dfR121))
*  Render notification content with `role` attribute based on `role` prop or default to `alert` ([link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccR57), [link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-8b66e26e1dc3ab97c7faf1e4f2d015c5b42d580fce0dd4c3043b7253bcdb83ccL71-R73))
*  Document `role` prop in English and Chinese API sections of `index.en-US.md` and `index.zh-CN.md` files ([link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-9b1700ccbd4da0f60742ae024592c27e0118d7307f5a9ed69a91c560c374749fR59), [link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-9954e364a4a4121f77d3f9f87225ef361d39c004720443dd896bb31cfe3c6d22R60))
*  Test that notification component renders `role` attribute correctly in `index.test.tsx` file ([link](https://github.com/ant-design/ant-design/pull/42484/files?diff=unified&w=0#diff-ed8976c7e832d9befa4199cc192b025c91cb3ff525a3649834d7545fdd89f690R304-R315))

### CODE REVIEW

File: components/notification/index.zh-CN.md [hunk 1]
```diff
@@ -57,6 +57,7 @@ config 参数如下：
 | message | 通知提醒标题，必选 | ReactNode | - |
 | placement | 弹出位置，可选 `topLeft` `topRight` `bottomLeft` `bottomRight` | string | `topRight` |
 | style | 自定义内联样式 | [CSSProperties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e434515761b36830c3e58a970abf5186f005adac/types/react/index.d.ts#L794) | - |
+| role | 供屏幕阅读器识别的通知内容语义，默认为 `alert`。此情况下屏幕阅读器会立即打断当前正在阅读的其他内容，转而阅读通知内容 | `alert \| status` | `alert` |
```
Comment (li-jia-nan) [inline components/notification/index.zh-CN.md:60]: 后面需要加上版本号 `5.6.0`

File: components/notification/index.en-US.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ The properties of config are as follows:
 | message | The title of notification box (required) | ReactNode | - |
 | placement | Position of Notification, can be one of `topLeft` `topRight` `bottomLeft` `bottomRight` | string | `topRight` |
 | style | Customized inline style | [CSSProperties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e434515761b36830c3e58a970abf5186f005adac/types/react/index.d.ts#L794) | - |
+| role | The semantics of notification content recognized by screen readers. The default value is `alert`. When set as the default value, the screen reader will promptly interrupt any ongoing content reading and prioritize the notification content for immediate attention. | `alert \| status` | `alert` |
```
Comment (li-jia-nan) [inline components/notification/index.en-US.md:59]: 后面需要加上版本号 `5.6.0`

File: components/notification/index.zh-CN.md (hunk 1 above)
Comment (guan404ming) [reply components/notification/index.zh-CN.md]: 已經加上了，再麻煩大佬了～

File: components/notification/index.en-US.md [hunk 3]
```diff
@@ -56,7 +56,7 @@ The properties of config are as follows:
 | message | The title of notification box (required) | ReactNode | - |
 | placement | Position of Notification, can be one of `topLeft` `topRight` `bottomLeft` `bottomRight` | string | `topRight` |
 | style | Customized inline style | [CSSProperties](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e434515761b36830c3e58a970abf5186f005adac/types/react/index.d.ts#L794) | - |
-| role | The semantics of notification content recognized by screen readers. The default value is `alert`. When set as the default value, the screen reader will promptly interrupt any ongoing content reading and prioritize the notification content for immediate attention. | `alert \| status` | `alert` |
+| role | The semantics of notification content recognized by screen readers. The default value is `alert`. When set as the default value, the screen reader will promptly interrupt any ongoing content reading and prioritize the notification content for immediate attention. | `alert \| status` | `alert` | 5.6.0 |
```
Comment (afc163) [inline components/notification/index.en-US.md:59]: https://preview-42484-ant-design.surge.sh/components/notification-cn#api

图片

Comment (guan404ming) [reply components/notification/index.en-US.md:59]: 已修正，再麻煩大佬了～

File: components/notification/PurePanel.tsx [hunk 4]
```diff
@@ -68,7 +70,7 @@ export function PureContent({
       className={classNames({
         [`${prefixCls}-with-icon`]: iconNode,
       })}
-      role="alert"
+      role={role || 'alert'}
```
Comment (li-jia-nan) [inline components/notification/PurePanel.tsx:73]:
```suggestion
      role={role}
```

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**